lib/gooddata/lcm/actions/collect_meta.rb

```diff
@@ -13,30 +13,43 @@ class CollectMeta < BaseAction
       with objects inside dashboards (reports, metrics ...) from development projects"
 
       PARAMS = define_params(self) do
-        description 'Tag Name'
+        description 'Production Tag Name'
         param :production_tag, instance_of(Type::StringType), required: false
 
         description 'Development Client Used for Connecting to GD'
         param :development_client, instance_of(Type::GdClientType), required: true
 
         description 'Synchronization Info'
         param :synchronize, array_of(instance_of(Type::SynchronizationInfoType)), required: true, generated: true
+
+        description 'Segments to search for segment-specific production tags'
+        param :segments, array_of(instance_of(Type::SegmentType)), required: false
       end
 
       class << self
         def call(params)
           results = []
 
           development_client = params.development_client
+          segments_to_tags = Helpers.segment_production_tags(params.segments)
 
           synchronize = params.synchronize.pmap do |info|
             from = info.from
             from_project = development_client.projects(from) || fail("Invalid 'from' project specified - '#{from}'")
 
-            if params.production_tag
-              objects = GoodData::Dashboard.find_by_tag(params.production_tag, project: from_project, client: development_client)
+            segment_tags = segments_to_tags[info.segment]
+            production_tags = Helpers.parse_production_tags(params.production_tag, segment_tags)
```

Comment:
I think we have 2 ways to specify `production_tag`: in global scope and in segment scope and the segment scope will override the global one. But looks like you collect them all. IHOP, if we specify `production_tag` in segment scope, it will always override the one in global. So we can easy keep track the tag to transfer and do not transfer a wrong tag by mistake because we forget to remove the `production_tag` in global scope.